compose/ui/ui/src/uikitInstrumentedTest/kotlin/android/compose/ui/test/UIKitInstrumentedTest.kt

```diff
@@ -0,0 +1,205 @@
+                    val placeable = measurables.firstOrNull()?.measure(constraints)
+
+                    val width = placeable?.width ?: constraints.minWidth
+                    val height = placeable?.height ?: constraints.minHeight
+
+                    layout(width, height) {
+                        placeable?.placeRelative(0, 0)
+                    }
+                }
+            )
+        } as ComposeContainer
+        window.rootViewController = composeContainer
+        window.makeKeyAndVisible()
+        waitUntil { onDraw }
+    }
+
+    private fun keyboardUserInfo(height: Dp, animated: Boolean): Map<Any?, *> {
+        val keyboardHeight = screenSize.height - height
+        val keyboardDuration = if (animated) 0.25 else 0.0
+        return mapOf(
+            UIKeyboardAnimationDurationUserInfoKey to NSNumber(double = keyboardDuration),
+            UIKeyboardFrameEndUserInfoKey to NSValue.valueWithCGRect(
+                CGRectMake(
+                    x = 0.0,
+                    y = keyboardHeight.value.toDouble(),
+                    width = screenSize.width.value.toDouble(),
+                    height = keyboardHeight.value.toDouble()
+                )
+            ),
+            UIKeyboardAnimationCurveUserInfoKey to NSNumber(unsignedInteger = 0uL)
+        )
+    }
+
+    fun showKeyboard(height: Dp = 216.dp, animated: Boolean = true) {
+        NSNotificationCenter.defaultCenter().postNotification(
+            NSNotification(
+                name = UIKeyboardWillShowNotification,
+                `object` = null,
+                userInfo = keyboardUserInfo(height, animated)
+            )
+        )
+        keyboardHeight = height
+        resizeKeyboard(height, animated)
+    }
+
+    fun resizeKeyboard(newHeight: Dp, animated: Boolean = true) {
+        NSNotificationCenter.defaultCenter().postNotification(
+            NSNotification(
+                name = UIKeyboardWillChangeFrameNotification,
+                `object` = null,
+                userInfo = keyboardUserInfo(newHeight, animated)
+            )
+        )
+        keyboardHeight = newHeight
+    }
+
+    fun hideKeyboard(animated: Boolean = true) {
+        NSNotificationCenter.defaultCenter().postNotification(
+            NSNotification(
+                name = UIKeyboardWillChangeFrameNotification,
+                `object` = null,
+                userInfo = keyboardUserInfo(0.dp, animated)
+            )
+        )
+        keyboardHeight = 0.dp
+    }
+
+    private val isIdle: Boolean get() {
+        val hadSnapshotChanges = Snapshot.current.hasPendingChanges()
+        val hasPendingRecomposer = Recomposer.runningRecomposers.value.any {
+            it.hasPendingWork
+        }
+        val isApplyObserverNotificationPending = Snapshot.isApplyObserverNotificationPending
+
+        return !hadSnapshotChanges && !hasPendingRecomposer && !isApplyObserverNotificationPending
```

Comment:
Why is this different from how `SkikoComposeUiTest.isIdle()` does it?
1. SkikoComposeUiTest doesn't look at Recomposer.runningRecomposers.
2. This `isIdle` doesn't look at `PlatformRootForTest.hasPendingMeasureOrLayout`.
3. This `isIdle` doesn't look at `ComposeScene.hasInvalidations`

Reply:
The tests we're running and SkikoComposeUiTest have bit different goals and setup configuration. Because of that the decisions where made.
1. The check came from here: https://github.com/JetBrains/compose-multiplatform-core/blob/77a3ddce223a97268f53b35d2dcb99836bb95b87/compose/ui/ui-test/src/androidMain/kotlin/androidx/compose/ui/test/ComposeIdlingResource.android.kt#L64
2. It's not easy to setup `PlatformRootForTest`, and it does not make much sense for iOS tests.
3. `hasInvalidations()` can be nice replacement for `Recomposer.runningRecomposers`. Let me check.